tests/test_artifactory_commands.py

```diff
@@ -0,0 +1,102 @@
+import os
+import tempfile
+
+from tools import run
+
+try:
+    # to test locally with an artifactory instance that already has
+    # extensions-stg and extensions-prod repos, define this dict of credentials
+    # in credentials.py (the file is gitignored)
+
+    # environment = {
+    #     "CONAN_LOGIN_USERNAME_EXTENSIONS_PROD": "......",
+    #     "CONAN_PASSWORD_EXTENSIONS_PROD": "......",
+    #     "CONAN_LOGIN_USERNAME_EXTENSIONS_STG": "......",
+    #     "CONAN_PASSWORD_EXTENSIONS_STG": "......",
+    #     "ART_URL": "https://url/artifactory",
+    # }
+
+    from credentials import environment
+except ImportError:
+    environment = {}
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def conan_test():
+    old_env = dict(os.environ)
+    env_vars = {"CONAN_HOME": tempfile.mkdtemp(suffix='conans')}
+    os.environ.update(env_vars)
+    os.environ.update(environment)
+    current = tempfile.mkdtemp(suffix="conans")
+    cwd = os.getcwd()
+    os.chdir(current)
+    run("conan profile detect")
+    run(f'conan remote add extensions-prod {os.getenv("ART_URL")}/api/conan/extensions-prod')
+    run(f'conan remote add extensions-stg {os.getenv("ART_URL")}/api/conan/extensions-stg')
+
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+        os.environ.clear()
+        os.environ.update(old_env)
+
+
+def test_build_info_create():
+    repo = os.path.join(os.path.dirname(__file__), "..")
+
+    build_name = "mybuildinfo"
+    build_number = "1"
+
+    run(f"conan config install {repo}")
+    run("conan new cmake_lib -d name=mypkg -d version=1.0 --force")
+
+    run("conan create . --format json -tf='' -s build_type=Release > create_release.json")
+    run("conan create . --format json -tf='' -s build_type=Debug > create_debug.json")
+
+    run("conan remove mypkg* -c -r extensions-stg")
+    run("conan remove mypkg* -c -r extensions-prod")
+
+    run("conan upload mypkg/1.0 -c -r extensions-stg")
+
+    run(f'conan art:build-info create create_release.json {build_name}_release {build_number} > {build_name}_release.json')
```

Comment:
Very good observation, yes, it requires the upload because we need to have the artifacts that are uploaded to Artifactory in the local cache and that happens in the upload process. I had a [TODO](https://github.com/conan-io/conan-extensions/blob/d6623f3991d5f66b6e5cb8574e721d0b2198856f/extensions/commands/art/cmd_build_info.py#L101) to raise in case the upload had not been made before this, I'll add the code to do it and some comments.

Reply:
Added the `test_fail_if_not_uploaded` test